plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -979,6 +980,22 @@ public String quoted(RemoteTableName remoteTableName)
         return emptyMap();
     }
 
+    @Override
+    public OptionalLong delete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        QueryBuilder queryBuilder = new QueryBuilder(this);
+        JdbcTableHandle tableHandle = (JdbcTableHandle) handle;
+        checkArgument(tableHandle.getGroupingSets().isEmpty() && tableHandle.getLimit().isEmpty(), "Unable to delete from synthetic table: %s", tableHandle);
+        try (Connection connection = connectionFactory.openConnection(session);
+                PreparedStatement preparedStatement = queryBuilder.buildDeleteSql(session, connection, tableHandle.getRemoteTableName(), tableHandle.getConstraint())) {
+            connection.setAutoCommit(true);
```

Comment:
isn't it default for JDBC?

Reply:
It looks it is not.

Reply:
For sure it is default for pooled connection factory.

I think we have to separate openConnection to two methods: openReadOnlyConnection(readonly=true, autocommit=false) and openConnectionForWrite(readonly=false, autocommit=true). But not in this PR.

Reply:
I think we have to separate openConnection to two methods: openReadOnlyConnection(readonly=true, autocommit=false) and openConnectionForWrite(readonly=false, autocommit=true). But not in this PR.